src/tokenizer/ngram_tokenizer.rs

```diff
@@ -0,0 +1,173 @@
+use super::{Token, TokenStream, Tokenizer};
+
+/// Tokenize the text by splitting words into n-grams of the given size(s)
+///
+/// Example: `hello` would be tokenized as (min_gram: 2, max_gram: 3)
+///
+/// | Term     | he  | hel | el  | ell | ll  | llo | lo |
+/// |----------|-----|-----|-----|-----|-----|-----|----|
+/// | Position | 0   | 0   | 1   | 1   | 2   | 2   | 3  |
+/// | Offsets  | 0,2 | 0,3 | 1,3 | 1,4 | 2,4 | 2,5 | 3,5|
+///
+/// # Example
+///
+/// ```
+/// extern crate tantivy;
+/// use tantivy::tokenizer::*;
+///
+/// fn assert_token(token: &Token, position: usize, text: &str, from: usize, to: usize) {
+///   assert_eq!(
+///       token.position, position,
+///       "expected position {} but {:?}",
+///       position, token
+///   );
+///   assert_eq!(token.text, text, "expected text {} but {:?}", text, token);
+///   assert_eq!(
+///       token.offset_from, from,
+///       "expected offset_from {} but {:?}",
+///       from, token
+///   );
+///   assert_eq!(
+///       token.offset_to, to,
+///       "expected offset_to {} but {:?}",
+///       to, token
+///   );
+/// }
+/// # fn main() {
+/// let tokenizer_manager = TokenizerManager::default();
```

Comment:
You can leave it as is, but this is a bit noisy for a doctest.

Reply:
Yeah, I hear that. :/ I'm not a huge fan, but not sure how to best expose the desired output. I pulled the `assert_token` out of the `#[cfg(test)]` so that I could reference it in the doctest. That helped a little bit. I'm thinking of trying to build some common collector as well, to reduce the noise and around that. Hmm.